### DISCUSSION TITLE
i want to reshape this dataset using r programming

### DISCUSSION THREAD

**Discussion by adedayosam** [microsoft/ML-For-Beginners#731] [PAT - Progress Assessment Tools]
[data.csv](https://github.com/microsoft/ML-For-Beginners/files/15026221/data.csv)

**Comment by DylanPrinsloo** [comment]
Depends on the set, however I have gone through the liberty of helping out!

## **Wide to long:**

data_long <- reshape(data_wide, varying = list(c("time1", "time2")), 
                     v.names = "time", times = c(1, 2), 
                     direction = "long")

## **Long to wide**
data_wide <- reshape(data_long, idvar = "id", timevar = "time", 
                     direction = "wide")

This is not all of it, but you will get the just of it and a key insight would be to use the Docs. 
Ai is also pretty useful for beginner understanding.

**Reply by Hamza-farid** [reply]
That’s a really helpful example thanks for sharing! To add on: the reshape() function in R is indeed the classic way to go between wide and long formats. For beginners, the modern tidyr package (pivot_longer() and pivot_wider()) often feels a bit easier and more readable, but both approaches are valuable depending on the workflow.

And you’re absolutely right checking the docs or using AI tools to clarify the syntax can really speed up the learning curve.
